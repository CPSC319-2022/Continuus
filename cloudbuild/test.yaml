steps:
  - name: 'gcr.io/cloud-builders/docker'
    entrypoint: 'bash'
    args: ['-c', 'docker pull northamerica-northeast2-docker.pkg.dev/automatic-bot-376307/continuus/$SHORT_SHA-test']

  - name: 'gcr.io/cloud-builders/docker'
    entrypoint: 'bash'
    args: ['-c', 'docker run --rm northamerica-northeast2-docker.pkg.dev/automatic-bot-376307/continuus/$SHORT_SHA-test npm run test']

options:
  logging: CLOUD_LOGGING_ONLY
