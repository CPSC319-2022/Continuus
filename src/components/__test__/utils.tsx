import React, { type PropsWithChildren } from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { configureStore, type PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import type { AppStore, RootState } from "~/redux/store";
import { reducer } from "~/redux/slices";
// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: reducer, preloadedState }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
