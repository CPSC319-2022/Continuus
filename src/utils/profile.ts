/**
 * Returns a path to the profile page for the given profile. If the argument is undefined, redirect to the home page.
 *
 * @param profile the user id of the account
 *
 * @return the profile path if the profile is a string, otherwise the home path
 */
export const userPathToProfile = (profile: string | undefined): string => {
    if (profile === undefined) {
        return '/';
    }

    return '/profile/' + encodeURIComponent(profile);
}
