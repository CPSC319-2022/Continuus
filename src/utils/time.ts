export const timeAgo = (inputDate: Date): string => {
  const now = new Date();
  const diffInSeconds = Math.max(Math.floor(
    (now.getTime() - inputDate.getTime()) / 1000
  ), 0);

  if (diffInSeconds < 60) {
    return `${diffInSeconds} ${diffInSeconds === 1 ? "second" : "seconds"} ago`;
  }

  const diffInMinutes = Math.floor(diffInSeconds / 60);
  if (diffInMinutes < 60) {
    return `${diffInMinutes} ${diffInMinutes === 1 ? "minute" : "minutes"} ago`;
  }

  const diffInHours = Math.floor(diffInMinutes / 60);
  if (diffInHours < 24) {
    return `${diffInHours} ${diffInHours === 1 ? "hour" : "hours"} ago`;
  }

  const diffInDays = Math.floor(diffInHours / 24);
  if (diffInDays < 30) {
    return `${diffInDays} ${diffInDays === 1 ? "day" : "days"} ago`;
  }

  const diffInMonths = Math.floor(diffInDays / 30);
  if (diffInMonths < 12) {
    return `${diffInMonths} ${diffInMonths === 1 ? 'month' : 'months'} ago`;
  }

  const diffInYears = Math.floor(diffInMonths / 12);
  return `${diffInYears} ${diffInYears === 1 ? 'year' : 'years'} ago`;
};
