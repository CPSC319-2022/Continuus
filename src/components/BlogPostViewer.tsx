import type { BlogPost, User, Comment } from "@prisma/client";
import { useState, useMemo, useEffect } from "react";
import { useInView } from "react-intersection-observer";
import { api } from "~/utils/api";
import { CommentModal } from "./CommentModal";
import { Spinner } from "./Spinner";
import { BlogPost as BlogPostComponent } from "./BlogPost";

type Post = BlogPost & { user: User; comments: (Comment & { user: User })[] };

export const BlogPostViewer: React.FC = () => {
  const [view, setView] = useState<string>("Recent");
  const {
    data: blogPosts,
    fetchNextPage,
    hasNextPage,
  } = api.blogPost.get.useInfiniteQuery(
    {
      take: 20,
    },
    { getNextPageParam: (lastPage) => ({ id: lastPage.nextCursor }) }
  );

  const posts = useMemo(
    () =>
      blogPosts?.pages.reduce(
        (result, curr) => [...result, ...curr.items],
        [] as Post[]
      ),
    [blogPosts]
  );

  const { ref, inView } = useInView();

  useEffect(() => {
    if (inView) {
      void (async () => {
        await fetchNextPage();
      })();
    }
  }, [inView, fetchNextPage]);

  return (
    <div className="w-full md:w-1/2 ">
      <div className="mb-6 flex w-full justify-end">
        <select
          className="h-8 w-64 max-w-xs border-b-2 bg-white"
          onChange={(e) => setView(e.target.value)}
        >
          <option>Recent</option>
          <option>Popular</option>
        </select>
      </div>
      {posts?.length === 0
        ? "Nothing to see here"
        : posts?.map(
            ({
              id,
              title,
              updatedAt,
              content,
              comments,
              user: { name, image },
            }) => (
              <>
                <div key={id} className="mb-6">
                  <BlogPostComponent
                    id={id}
                    name={name as string}
                    title={title}
                    lastUpdated={updatedAt}
                    imageUrl={image as string}
                    content={content}
                    comments={(comments as Comment[]).length}
                  />
                </div>
                <CommentModal
                  id={id}
                  title={title}
                  comments={comments as (Comment & { user: User })[]}
                  poster={name as string}
                  lastUpdated={updatedAt}
                  post={content}
                  posterAvatarUrl={image as string}
                />
              </>
            )
          )}
      {hasNextPage && (
        <div className="my-5 flex justify-center" ref={ref}>
          <Spinner size={2} />
        </div>
      )}
    </div>
  );
};
