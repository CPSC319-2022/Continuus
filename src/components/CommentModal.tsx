import { Comment } from "./Comment";
import React, { useState } from "react";
import ReactMarkdown from "react-markdown";
import remarkGfm from "remark-gfm";
import remarkSlug from "remark-slug";
import type { Comment as CommentType, User } from "@prisma/client";
import { timeAgo } from "~/utils/time";
import { ProfilePicture } from "./ProfilePicture";
import { BlogPostActionsMenu } from "~/components/BlogPostActionsMenu";
import { api } from "~/utils/api";
import { useSession } from "next-auth/react";
import { shouldSeeActions, isAuthor } from "~/components/util";
import {userPathToProfile} from "~/utils/profile";

type CommentEntry = CommentType & { user: User };

export interface CommentModalProps {
  id: string;
  title: string;
  poster: string;
  lastUpdated: Date;
  createdAt: Date;
  post: string;
  comments: CommentEntry[];
  content: string;
  authorId: string;
  imgUrl?: string | null;
}

export const CommentModal: React.FC<CommentModalProps> = ({
  id,
  title,
  poster,
  lastUpdated,
  createdAt,
  post,
  comments,
  content,
  authorId,
  imgUrl,
}) => {
  const [input, setInput] = useState<string>("");

  const utils = api.useContext();

  const currUser = api.user.currentUser.useQuery();

  const { status } = useSession();

  const createCommentMutation = api.comment.create.useMutation({
    onSuccess() {
        return Promise.all([utils.blogPost.get.invalidate(),
                            utils.comment.count.invalidate()]);
    },
  });

  const userId = currUser?.data?.id;

  const handleCommentChange = (event: {
    target: { value: React.SetStateAction<string> };
  }) => {
    setInput(event.target.value);
  };

  const handlePostButtonClick = () => {
    createCommentMutation.mutate({
      data: {
        userId: userId!,
        blogPostId: id,
        content: input,
      },
    });

    setInput("");
  };

  return (
    <>
      <input type="checkbox" id={`modal-${id}`} className="modal-toggle" />
      <label htmlFor={`modal-${id}`} className="modal z-10 cursor-pointer">
        <label
          className="card-body modal-box relative m-[-10px] w-11/12 max-w-5xl rounded-md"
          htmlFor=""
        >
          <div className="mb-3 flex w-full justify-between">
            <div className="flex">
              <div className="avatar self-center">
                  <ProfilePicture 
                      size={2.5}
                      redirectLink={userPathToProfile(authorId)} 
                      imgUrl={imgUrl}
                  />
              </div>
              <div className="ml-3">
                <p className="text-lg font-bold">{poster}</p>
                <p className="text-sm text-slate-400">{`${timeAgo(createdAt)}${
                  createdAt.getTime() !== lastUpdated.getTime()
                    ? ` (updated ${timeAgo(lastUpdated)})`
                    : ""
                }`}</p>
              </div>
            </div>
            {
              shouldSeeActions(status, currUser.data, authorId) && (
              <BlogPostActionsMenu
                id={id}
                title={title}
                content={content}
                isAuthor={isAuthor(currUser.data, authorId)}
              />
            )}
          </div>
          <p className="mb-3 text-xl font-bold">{title}</p>
          <div className="prose max-w-none ">
            <ReactMarkdown remarkPlugins={[remarkGfm, remarkSlug]}>
              {post}
            </ReactMarkdown>
          </div>
          <div className="self-end">
            <p className="btn-link text-highlight-green no-underline hover:no-underline">
              {comments.length} Comments
            </p>
          </div>
          <div>
            {comments.map(
              ({
                id: commentId,
                content: comment,
                createdAt: dateAdded,
                user: { name: authorName, id: commenterId, image},
              }) => (
                <Comment
                  key={`${commentId}`}
                  dateAdded={dateAdded}
                  comment={comment}
                  name={authorName}
                  userId={commenterId}
                  imgUrl={image}
                />
              )
            )}
          </div>
          <div className="m-[2rem] mx-0 mb-0">
            <div
              className={`${userId ? "" : "tooltip tooltip-info"} w-full`}
              data-tip="Sign in to comment"
            >
              <input
                type="text"
                placeholder="Write a new comment"
                value={input}
                onChange={handleCommentChange}
                readOnly={userId ? false : true}
                disabled={userId ? false : true}
                className={`${
                  userId ? "" : "input-disabled cursor-not-allowed"
                }input-bordered mb-1 w-full rounded-sm border-[1px] p-2`}
              />
            </div>

            <div className="mb-[-10px] flex justify-end gap-2">
              <button
                className={`${
                  userId
                    ? " border-highlight-green bg-highlight-green "
                    : "input-disabled cursor-not-allowed bg-gray-400 text-black"
                } mt-[0.5rem] h-10 rounded-sm px-5`}
                onClick={handlePostButtonClick}
                disabled={userId ? false : true}
              >
                Post
              </button>
              <label
                className={`mt-[0.5rem] h-10 rounded-sm border-[2px] border-slate-200 bg-white px-5 pt-[0.38rem] text-center text-black hover:cursor-pointer`}
                htmlFor={`modal-${id}`}
              >
                Cancel
              </label>
            </div>
          </div>
        </label>
      </label>
    </>
  );
};
