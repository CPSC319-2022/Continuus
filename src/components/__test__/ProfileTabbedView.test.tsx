import { describe, it, expect, beforeEach, vi } from "vitest";
import { cleanup, render, screen, waitFor } from "@testing-library/react";
import { renderWithProviders } from "./utils";
import userEvent from "@testing-library/user-event";
import { adminUser, blogPosts } from "./BlogPostViewer.test";
import { mockComments } from "./CommentModal.test";

const props = {
  userId: "test-id",
  createdAt: new Date(),
  name: "test user",
  imgUrl: "image",
};

vi.mock("next-auth/react", () => ({
  useSession: () => ({
    data: {
      expires: "1",
      user: { id: "test-user-id", email: "a", name: "hi", image: "c" },
    },
  }),
}));

const setup = () => {
  vi.doMock("~/utils/time", () => ({
    timeAgo: () => "1 second ago",
  }));

  vi.doMock("~/utils/api", () => ({
    api: {
      user: {
        currentUser: {
          useQuery: () => ({
            data: adminUser,
            isLoading: false,
          }),
        },
      },
      blogPost: {
        get: {
          useInfiniteQuery: () => ({
            data: blogPosts,
            fetchNextPage: vi.fn(),
            hasNextPage: false,
          }),
        },
        count: {
          useQuery: () => ({
            data: 5,
            isLoading: false,
          }),
        },
      },
      comment: {
        count: {
          useQuery: () => ({
            data: 5,
            isLoading: false,
          }),
        },
        get: {
          useInfiniteQuery: () => ({
            data: {
              pages: [
                {
                  items: mockComments.map((comment) => ({
                    ...comment,
                    blogPost: {
                      id: comment.blogPostId,
                      title: "test blog post",
                    },
                  })),
                },
              ],
            },
            fetchNextPage: vi.fn(),
            hasNextPage: false,
            status: null,
          }),
        },
        getUnique: {
          useQuery: () => ({
            data: {
              id: "comment-1",
              blogPostId: "post-1",
              content: "test comment",
              createdAt: new Date(),
              updatedAt: new Date(),
              user: {
                name: "test user 2",
                id: "user-2",
                image: "image",
              },
            },
            isLoading: false,
          }),
        },
        update: {
          useMutation: () => ({
            onSuccess: vi.fn(),
          }),
        },
        delete: {
          useMutation: () => ({
            onSuccess: vi.fn(),
          }),
        },
      },
      useContext: () => ({
        user: {
          invalidate: vi.fn(),
        },
      }),
    },
  }));
};

describe("Profile tabbed view: Snapshot", () => {
  beforeEach(() => {
    cleanup();
    vi.clearAllMocks();
    vi.resetModules();
  });

  it("should render profile tabbed view in blog post mode", async () => {
    setup();
    const { ProfileTabbedView } = await import(
      "~/components/ProfileTabbedView"
    );
    const body = renderWithProviders(
      <ProfileTabbedView {...props} />
    ).baseElement;
    expect(body).toMatchSnapshot();
  });

  it("should render profile tabbed view in comment mode", async () => {
    setup();
    const { ProfileTabbedView } = await import(
      "~/components/ProfileTabbedView"
    );
    const body = renderWithProviders(<ProfileTabbedView {...props} />);
    const commentTab = screen.getByTestId("tab-comment-viewer");
    await userEvent.click(commentTab);
    await waitFor(() => {
      expect(screen.getByTestId("comment-viewer")).toBeTruthy();
    });
    expect(body.baseElement).toMatchSnapshot();
  });
});
