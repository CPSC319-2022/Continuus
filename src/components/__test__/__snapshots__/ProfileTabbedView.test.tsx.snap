// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Blog post viewer: Snapshot > render no blog posts 1`] = `
<body>
  <div>
    <div
      class="w-full self-center"
    >
      Nothing to see here
    </div>
  </div>
</body>
`;

exports[`Blog post viewer: Snapshot > render with blog posts 1`] = `
<body>
  <div>
    <div
      class="w-full self-center"
    >
      <div
        class="mb-6"
      >
        <div
          class="relative"
        >
          <div
            class="card w-full rounded-md shadow-md border hover:cursor-pointer hover:border-highlight-green transition-all"
          >
            <div
              class="card-body"
            >
              <div
                class="mb-3 flex w-full justify-between"
              >
                <div
                  class="flex"
                >
                  <div
                    class="avatar self-center hover:scale-110 transition-all"
                  >
                     
                    <a
                      href="/profile/user-1"
                    >
                      <img
                        alt="avatar"
                        class="inline-block aspect-square rounded-[50%] "
                        src="imageUrl"
                        style="width: 2.5rem;"
                      />
                    </a>
                  </div>
                  <div
                    class="ml-3"
                  >
                    <a
                      href="/profile/user-1"
                    >
                      <p
                        class="text-lg font-bold hover:underline"
                      >
                        test user
                      </p>
                    </a>
                    <p
                      class="text-sm text-gray-400"
                    >
                      X seconds ago
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="mb-3 w-fit text-2xl font-bold"
              >
                title
              </div>
              <div
                class="prose relative"
              >
                <p>
                  content
                </p>
                <div />
              </div>
              <div
                class="self-end"
              >
                <p
                  class="text-highlight-green no-underline"
                >
                  1
                   Comments
                </p>
              </div>
            </div>
          </div>
          <div
            class="absolute right-10 top-12"
          />
        </div>
      </div>
      <div
        class="mb-6"
      >
        <div
          class="relative"
        >
          <div
            class="card w-full rounded-md shadow-md border hover:cursor-pointer hover:border-highlight-green transition-all"
          >
            <div
              class="card-body"
            >
              <div
                class="mb-3 flex w-full justify-between"
              >
                <div
                  class="flex"
                >
                  <div
                    class="avatar self-center hover:scale-110 transition-all"
                  >
                     
                    <a
                      href="/profile/user-3"
                    >
                      <img
                        alt="avatar"
                        class="inline-block aspect-square rounded-[50%] "
                        src="anotherImageUrl"
                        style="width: 2.5rem;"
                      />
                    </a>
                  </div>
                  <div
                    class="ml-3"
                  >
                    <a
                      href="/profile/user-3"
                    >
                      <p
                        class="text-lg font-bold hover:underline"
                      >
                        another test user
                      </p>
                    </a>
                    <p
                      class="text-sm text-gray-400"
                    >
                      X seconds ago
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="mb-3 w-fit text-2xl font-bold"
              >
                another title
              </div>
              <div
                class="prose relative"
              >
                <p>
                  another content
                </p>
                <div />
              </div>
              <div
                class="self-end"
              >
                <p
                  class="text-highlight-green no-underline"
                >
                  1
                   Comments
                </p>
              </div>
            </div>
          </div>
          <div
            class="absolute right-10 top-12"
          />
        </div>
      </div>
      <div
        class="mb-6"
      >
        <div
          class="relative"
        >
          <div
            class="card w-full rounded-md shadow-md border hover:cursor-pointer hover:border-highlight-green transition-all"
          >
            <div
              class="card-body"
            >
              <div
                class="mb-3 flex w-full justify-between"
              >
                <div
                  class="flex"
                >
                  <div
                    class="avatar self-center hover:scale-110 transition-all"
                  >
                     
                    <a
                      href="/profile/user-5"
                    >
                      <img
                        alt="avatar"
                        class="inline-block aspect-square rounded-[50%] "
                        src="differentImageUrl"
                        style="width: 2.5rem;"
                      />
                    </a>
                  </div>
                  <div
                    class="ml-3"
                  >
                    <a
                      href="/profile/user-5"
                    >
                      <p
                        class="text-lg font-bold hover:underline"
                      >
                        different test user
                      </p>
                    </a>
                    <p
                      class="text-sm text-gray-400"
                    >
                      X seconds ago
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="mb-3 w-fit text-2xl font-bold"
              >
                different title
              </div>
              <div
                class="prose relative"
              >
                <p>
                  different content
                </p>
                <div />
              </div>
              <div
                class="self-end"
              >
                <p
                  class="text-highlight-green no-underline"
                >
                  1
                   Comments
                </p>
              </div>
            </div>
          </div>
          <div
            class="absolute right-10 top-12"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Comment Modal: Snapshots > render comment modal logged in with selected post 1`] = `
<body>
  <div />
</body>
`;

exports[`Comment Modal: Snapshots > render comment modal not logged in no selected post 1`] = `
<body>
  <div />
</body>
`;

exports[`Comment Modal: Snapshots > render comment modal not logged in with selected post 1`] = `
<body>
  <div />
</body>
`;

exports[`Profile tabbed view: Snapshot > should render profile tabbed view in blog post mode 1`] = `
<body>
  <div>
    <div
      class="mt-4 flex min-h-screen w-full flex-col content-center items-center md:px-0"
    >
      <div
        class="p-8 mt-24"
      >
        <div
          class="relative"
        >
          <div
            class="w-36 h-36 mx-auto rounded-full shadow-md absolute inset-x-0 top-0 -mt-24 flex items-center justify-center"
          >
             
            <img
              alt="avatar"
              class="inline-block aspect-square rounded-[50%] "
              src="image"
              style="width: 40rem;"
            />
          </div>
        </div>
        <div
          class="mt-20 text-center pb-8"
        >
          <h1
            class="text-4xl font-bold"
          >
            test user
          </h1>
          <p
            class="mt-3 text-gray-400"
          >
            joined 
            <span
              class="text-grey-400"
            >
              1 second ago
            </span>
          </p>
        </div>
        <div
          class="wx-1/2"
        >
          <div
            class="grid grid-cols-2 gap-48 text-center grid-padding-64"
          >
            <div>
              <p
                class="font-bold text-gray-700 text-xl"
              >
                5
              </p>
              <p
                class="text-gray-400"
              >
                Blog Posts
              </p>
            </div>
            <div>
              <p
                class="font-bold text-gray-700 text-xl"
              >
                5
              </p>
              <p
                class="text-gray-400"
              >
                Comments
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="md:w-1/3 md:min-w-[40rem] w-11/12 relative"
        data-rttabs="true"
      >
        <ul
          class="mb-3 flex flex-row justify-start"
          role="tablist"
        >
          <li
            aria-controls="panel:r0:0"
            aria-disabled="false"
            aria-selected="true"
            class="w-1/2 md:w-32 cursor-pointer border-b-4 border-solid border-gray-200 p-4 text-center transition-all hover:border-b-highlight-green hover:font-bold border-b-4 border-solid border-b-highlight-green font-bold"
            data-rttab="true"
            data-testid="tab-post-viewer"
            id="tab:r0:0"
            role="tab"
            tabindex="0"
          >
            Blog Posts
          </li>
          <li
            aria-controls="panel:r0:1"
            aria-disabled="false"
            aria-selected="false"
            class="w-1/2 md:w-32 cursor-pointer border-b-4 border-solid border-gray-200 p-4 text-center transition-all hover:border-b-highlight-green hover:font-bold"
            data-rttab="true"
            data-testid="tab-comment-viewer"
            id="tab:r0:1"
            role="tab"
          >
            Comments
          </li>
        </ul>
        <div
          aria-labelledby="tab:r0:0"
          class="react-tabs__tab-panel react-tabs__tab-panel--selected"
          id="panel:r0:0"
          role="tabpanel"
        >
          <div
            class="w-full self-center"
          >
            <div
              class="mb-6"
            >
              <div
                class="relative"
              >
                <div
                  class="card w-full rounded-md shadow-md border hover:cursor-pointer hover:border-highlight-green transition-all"
                >
                  <div
                    class="card-body"
                  >
                    <div
                      class="mb-3 flex w-full justify-between"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="avatar self-center hover:scale-110 transition-all"
                        >
                           
                          <a
                            href="/profile/user-1"
                          >
                            <img
                              alt="avatar"
                              class="inline-block aspect-square rounded-[50%] "
                              src="imageUrl"
                              style="width: 2.5rem;"
                            />
                          </a>
                        </div>
                        <div
                          class="ml-3"
                        >
                          <a
                            href="/profile/user-1"
                          >
                            <p
                              class="text-lg font-bold hover:underline"
                            >
                              test user
                            </p>
                          </a>
                          <p
                            class="text-sm text-gray-400"
                          >
                            1 second ago
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="mb-3 w-fit text-2xl font-bold"
                    >
                      title
                    </div>
                    <div
                      class="prose relative"
                    >
                      <p>
                        content
                      </p>
                      <div />
                    </div>
                    <div
                      class="self-end"
                    >
                      <p
                        class="text-highlight-green no-underline"
                      >
                        1
                         Comments
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="absolute right-10 top-12"
                />
              </div>
            </div>
            <div
              class="mb-6"
            >
              <div
                class="relative"
              >
                <div
                  class="card w-full rounded-md shadow-md border hover:cursor-pointer hover:border-highlight-green transition-all"
                >
                  <div
                    class="card-body"
                  >
                    <div
                      class="mb-3 flex w-full justify-between"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="avatar self-center hover:scale-110 transition-all"
                        >
                           
                          <a
                            href="/profile/user-3"
                          >
                            <img
                              alt="avatar"
                              class="inline-block aspect-square rounded-[50%] "
                              src="anotherImageUrl"
                              style="width: 2.5rem;"
                            />
                          </a>
                        </div>
                        <div
                          class="ml-3"
                        >
                          <a
                            href="/profile/user-3"
                          >
                            <p
                              class="text-lg font-bold hover:underline"
                            >
                              another test user
                            </p>
                          </a>
                          <p
                            class="text-sm text-gray-400"
                          >
                            1 second ago
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="mb-3 w-fit text-2xl font-bold"
                    >
                      another title
                    </div>
                    <div
                      class="prose relative"
                    >
                      <p>
                        another content
                      </p>
                      <div />
                    </div>
                    <div
                      class="self-end"
                    >
                      <p
                        class="text-highlight-green no-underline"
                      >
                        1
                         Comments
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="absolute right-10 top-12"
                />
              </div>
            </div>
            <div
              class="mb-6"
            >
              <div
                class="relative"
              >
                <div
                  class="card w-full rounded-md shadow-md border hover:cursor-pointer hover:border-highlight-green transition-all"
                >
                  <div
                    class="card-body"
                  >
                    <div
                      class="mb-3 flex w-full justify-between"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="avatar self-center hover:scale-110 transition-all"
                        >
                           
                          <a
                            href="/profile/user-5"
                          >
                            <img
                              alt="avatar"
                              class="inline-block aspect-square rounded-[50%] "
                              src="differentImageUrl"
                              style="width: 2.5rem;"
                            />
                          </a>
                        </div>
                        <div
                          class="ml-3"
                        >
                          <a
                            href="/profile/user-5"
                          >
                            <p
                              class="text-lg font-bold hover:underline"
                            >
                              different test user
                            </p>
                          </a>
                          <p
                            class="text-sm text-gray-400"
                          >
                            1 second ago
                          </p>
                        </div>
                      </div>
                    </div>
                    <div
                      class="mb-3 w-fit text-2xl font-bold"
                    >
                      different title
                    </div>
                    <div
                      class="prose relative"
                    >
                      <p>
                        different content
                      </p>
                      <div />
                    </div>
                    <div
                      class="self-end"
                    >
                      <p
                        class="text-highlight-green no-underline"
                      >
                        1
                         Comments
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="absolute right-10 top-12"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          aria-labelledby="tab:r0:1"
          class="react-tabs__tab-panel"
          id="panel:r0:1"
          role="tabpanel"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Profile tabbed view: Snapshot > should render profile tabbed view in comment mode 1`] = `
<body>
  <div>
    <div
      class="mt-4 flex min-h-screen w-full flex-col content-center items-center md:px-0"
    >
      <div
        class="p-8 mt-24"
      >
        <div
          class="relative"
        >
          <div
            class="w-36 h-36 mx-auto rounded-full shadow-md absolute inset-x-0 top-0 -mt-24 flex items-center justify-center"
          >
             
            <img
              alt="avatar"
              class="inline-block aspect-square rounded-[50%] "
              src="image"
              style="width: 40rem;"
            />
          </div>
        </div>
        <div
          class="mt-20 text-center pb-8"
        >
          <h1
            class="text-4xl font-bold"
          >
            test user
          </h1>
          <p
            class="mt-3 text-gray-400"
          >
            joined 
            <span
              class="text-grey-400"
            >
              1 second ago
            </span>
          </p>
        </div>
        <div
          class="wx-1/2"
        >
          <div
            class="grid grid-cols-2 gap-48 text-center grid-padding-64"
          >
            <div>
              <p
                class="font-bold text-gray-700 text-xl"
              >
                5
              </p>
              <p
                class="text-gray-400"
              >
                Blog Posts
              </p>
            </div>
            <div>
              <p
                class="font-bold text-gray-700 text-xl"
              >
                5
              </p>
              <p
                class="text-gray-400"
              >
                Comments
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="md:w-1/3 md:min-w-[40rem] w-11/12 relative"
        data-rttabs="true"
      >
        <ul
          class="mb-3 flex flex-row justify-start"
          role="tablist"
        >
          <li
            aria-controls="panel:r1:0"
            aria-disabled="false"
            aria-selected="false"
            class="w-1/2 md:w-32 cursor-pointer border-b-4 border-solid border-gray-200 p-4 text-center transition-all hover:border-b-highlight-green hover:font-bold"
            data-rttab="true"
            data-testid="tab-post-viewer"
            id="tab:r1:0"
            role="tab"
          >
            Blog Posts
          </li>
          <li
            aria-controls="panel:r1:1"
            aria-disabled="false"
            aria-selected="true"
            class="w-1/2 md:w-32 cursor-pointer border-b-4 border-solid border-gray-200 p-4 text-center transition-all hover:border-b-highlight-green hover:font-bold border-b-4 border-solid border-b-highlight-green font-bold"
            data-rttab="true"
            data-testid="tab-comment-viewer"
            id="tab:r1:1"
            role="tab"
            tabindex="0"
          >
            Comments
          </li>
        </ul>
        <div
          aria-labelledby="tab:r1:0"
          class="react-tabs__tab-panel"
          id="panel:r1:0"
          role="tabpanel"
        />
        <div
          aria-labelledby="tab:r1:1"
          class="react-tabs__tab-panel react-tabs__tab-panel--selected"
          id="panel:r1:1"
          role="tabpanel"
        >
          <div
            data-testid="comment-viewer"
          >
            <div
              class="w-full mb-6 p-6 border rounded-md shadow-md"
            >
              <div
                class="mb-2 flex border-b"
              >
                <div
                  class="mb-2"
                >
                  Commented on
                </div>
                <div
                  class="mb-2 ml-1 font-bold cursor-pointer"
                >
                  test blog post
                </div>
              </div>
              <div
                class="relative m-3 rounded-md border-b p-4"
              >
                <div
                  class="flex"
                >
                  <div
                    class="avatar self-center transition-all hover:scale-110"
                  >
                     
                    <a
                      href="/"
                    >
                      <img
                        alt="avatar"
                        class="inline-block aspect-square rounded-[50%] "
                        src="image"
                        style="width: 2.5rem;"
                      />
                    </a>
                  </div>
                  <div
                    class="ml-3"
                  >
                    <a
                      href="/"
                    >
                      <p
                        class="text-lg font-bold hover:underline"
                      >
                        test user 2
                      </p>
                    </a>
                    <p
                      class="text-gray-400 text-sm"
                    >
                      1 second ago
                    </p>
                  </div>
                </div>
                <div
                  class="break-words"
                >
                  <p
                    class="mt-4 w-full"
                  >
                    test comment
                  </p>
                </div>
              </div>
            </div>
            <div
              class="w-full mb-6 p-6 border rounded-md shadow-md"
            >
              <div
                class="mb-2 flex border-b"
              >
                <div
                  class="mb-2"
                >
                  Commented on
                </div>
                <div
                  class="mb-2 ml-1 font-bold cursor-pointer"
                >
                  test blog post
                </div>
              </div>
              <div
                class="relative m-3 rounded-md border-b p-4"
              >
                <div
                  class="flex"
                >
                  <div
                    class="avatar self-center transition-all hover:scale-110"
                  >
                     
                    <a
                      href="/"
                    >
                      <img
                        alt="avatar"
                        class="inline-block aspect-square rounded-[50%] "
                        src="image"
                        style="width: 2.5rem;"
                      />
                    </a>
                  </div>
                  <div
                    class="ml-3"
                  >
                    <a
                      href="/"
                    >
                      <p
                        class="text-lg font-bold hover:underline"
                      >
                        test user 3
                      </p>
                    </a>
                    <p
                      class="text-gray-400 text-sm"
                    >
                      1 second ago
                    </p>
                  </div>
                </div>
                <div
                  class="break-words"
                >
                  <p
                    class="mt-4 w-full"
                  >
                    test comment
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
